### ISSUE TITLE
"use_socket: true" results in error "Recevived 0 instead of expected 1 bytes"

### ISSUE THREAD

**Issue by m0onspell** [php-amqplib/RabbitMqBundle#447]
I faced the exact same problem as described [here](https://stackoverflow.com/questions/39640489/error-reading-data-while-executing-a-rabbitmq-consumer-in-symfony).
Running in a docker container, `docker-compose.yml`:
```yml
    php:
        build: php
        volumes:
          - ../app/backend:/var/www/symfony
    rabbitmq:
          build: rabbitmq
          privileged: true
          ports:
            - "5672:5672"
            - "15672:15672"
            - "15671:15671"
          environment:
            - RABBITMQ_DEFAULT_USER=root
            - RABBITMQ_DEFAULT_PASS=default
          expose:
            - 5672
```
Symfony `config.yml`:
```yml
old_sound_rabbit_mq:
    connections:
        default:
            host:     'rabbitmq'
            port:     5672
            user:     'root'
            password: 'default'
            vhost:    '/'
            lazy:     true
            connection_timeout: 60
            read_write_timeout: 60

            # requires php-amqplib v2.4.1+ and PHP5.4+
            keepalive: false

            # requires php-amqplib v2.4.1+
            heartbeat: 30

            #requires php_sockets.dll
            use_socket: true # default false
    producers:
        offline_sync:
            connection:       default
            exchange_options: {name: 'offline-sync', type: direct}
            enable_logger: true
    consumers:
        offline_sync:
            connection:       default
            exchange_options: {name: 'offline-sync', type: direct}
            queue_options:    {name: 'offline-sync'}
            callback:         app.test_handler
            enable_logger: true
```
Running:
```
php bin/console rabbitmq:consumer -w offline_sync
```
Sockets and mbstring extensions are installed.
Setting `use_socket: false` fixes the issue.
So isn't it a bug? Are we unable to use sockets?

**Comment by m0onspell** [comment]
Creators don't care. Closing.

**Comment by webarchitect609** [comment]
@m0onspell , the issue has been fixed in [#448](https://github.com/php-amqplib/php-amqplib/pull/448) .
